kratos/tests/cpp_tests/processes/test_replace_elements_and_condition_process.cpp

```diff
@@ -27,233 +27,320 @@
+
+    std::vector<NodeType::Pointer> element_nodes_10 (4);
+    element_nodes_10[0] = p_node_9;
+    element_nodes_10[1] = p_node_12;
+    element_nodes_10[2] = p_node_11;
+    element_nodes_10[3] = p_node_8;
+    Tetrahedra3D4 <NodeType>::Pointer p_tetrahedra_10 = Kratos::make_shared<Tetrahedra3D4 <NodeType>>( PointerVector<NodeType>{element_nodes_10} );
+
+    std::vector<NodeType::Pointer> element_nodes_11 (4);
+    element_nodes_11[0] = p_node_3;
+    element_nodes_11[1] = p_node_2;
+    element_nodes_11[2] = p_node_1;
+    element_nodes_11[3] = p_node_6;
+    Tetrahedra3D4 <NodeType>::Pointer p_tetrahedra_11 = Kratos::make_shared<Tetrahedra3D4 <NodeType>>( PointerVector<NodeType>{element_nodes_11} );
+
+    Element::Pointer p_elem_0 = Kratos::make_shared<Element>(1, p_tetrahedra_0, p_elem_prop);
+    Element::Pointer p_elem_1 = Kratos::make_shared<Element>(2, p_tetrahedra_1, p_elem_prop);
+    Element::Pointer p_elem_2 = Kratos::make_shared<Element>(3, p_tetrahedra_2, p_elem_prop);
+    Element::Pointer p_elem_3 = Kratos::make_shared<Element>(4, p_tetrahedra_3, p_elem_prop);
+    Element::Pointer p_elem_4 = Kratos::make_shared<Element>(5, p_tetrahedra_4, p_elem_prop);
+    Element::Pointer p_elem_5 = Kratos::make_shared<Element>(6, p_tetrahedra_5, p_elem_prop);
+    Element::Pointer p_elem_6 = Kratos::make_shared<Element>(7, p_tetrahedra_6, p_elem_prop);
+    Element::Pointer p_elem_7 = Kratos::make_shared<Element>(8, p_tetrahedra_7, p_elem_prop);
+    Element::Pointer p_elem_8 = Kratos::make_shared<Element>(9, p_tetrahedra_8, p_elem_prop);
+    Element::Pointer p_elem_9 = Kratos::make_shared<Element>(10, p_tetrahedra_9, p_elem_prop);
+    Element::Pointer p_elem_10 = Kratos::make_shared<Element>(11, p_tetrahedra_10, p_elem_prop);
+    Element::Pointer p_elem_11 = Kratos::make_shared<Element>(12, p_tetrahedra_11, p_elem_prop);
+    this_model_part.AddElement(p_elem_0);
+    this_model_part.AddElement(p_elem_1);
+    this_model_part.AddElement(p_elem_2);
+    this_model_part.AddElement(p_elem_3);
+    this_model_part.AddElement(p_elem_4);
+    this_model_part.AddElement(p_elem_5);
+    this_model_part.AddElement(p_elem_6);
+    this_model_part.AddElement(p_elem_7);
+    this_model_part.AddElement(p_elem_8);
+    this_model_part.AddElement(p_elem_9);
+    this_model_part.AddElement(p_elem_10);
+    this_model_part.AddElement(p_elem_11);
+
+    // Compute process
+    Parameters settings( R"(
+    {
+        "element_name":"Element3D4N",
+        "condition_name": "SurfaceCondition3D3N"
+    }  )" );
+
+    ReplaceElementsAndConditionsProcess process(this_model_part, settings);
+    process.Execute();
+
+    // Same element type
+    std::string component_name;
+    for (auto& r_element : this_model_part.Elements()) {
+        CompareElementsAndConditionsUtility::GetRegisteredName(r_element, component_name);
+        KRATOS_CHECK_EQUAL(component_name, "Element3D4N");
+    }
+}
+
+/**
+* Checks the correct work of the nodal gradient compute
```

Comment:
I think this comment (and all of the comments before the functions in this file) refers to another test. I think it would be helpful for other developers to explain what you are testing here to know it in one glance.